src/deriving.mli

```diff
@@ -105,10 +105,12 @@ with type deriver := t
 
 val add :
   ?str_type_decl:(structure, rec_flag * type_declaration list) Generator.t ->
+  ?str_class_type_decl:(structure, class_type_declaration list) Generator.t ->
```

Comment:
This change could potentially break some rev deps, although I deem it quite unlikely.

I think it's fair to treat this as non breaking from our perspective given how the `add` and `add_alias` function are used in practice but still wanted for us to keep this in mind before we release this!

Reply:
Agreed. 

As a litmus test, I tested this change against 50 revdeps without any failures due to this. There were failures because of flakey tests and no solves.